text/0000-new-feature-checklist.md

```diff
@@ -0,0 +1,35 @@
+# New Feature Checklist
+
+## Summary
+
+Layout out a checklist of tasks to ensure the smooth addition of a new feature
+into a buildpack.
+
+## Motivation
+
+Currently there is no defined workflow for ensuring the smooth introduction of
+features into buildpacks. This means that steps often get overlooked and
+forgotten by maintainers. It also means that it is hard for new contributors to
+follow a smooth process and for them to understand what is fully required to
+introduce a new buildpack feature. Having something like a checklist would
+allow maintainers to become more consistent in enforcing a process for rolling
+out a new feature and will give new contributors a layout and plan for how they
+can go about adding a new feature that will be easily rolled out.
+
+## Implementation
+
+Add a NEW_FEATURE_CHECKLIST.md to the [community
+repository](https://github.com/paketo-buildpacks/community). The contents of
+that document should be as follows:
+```markdown
+- [ ] (If necessary) Open an RFC in the [RFCs repository](https://github.com/paketo-buildpacks/rfcs) to discuss the addition of the feature
+- [ ] (Optional) Open tracking issue in relevant buildpack repository
+- [ ] Open PR adding new feature to codebase
+    - [ ] (If necessary) Add integration test that exercises the new feature
+    - [ ] (If necessary) Update the README of the buildpack
+- [ ] Secure a release of the buildpack
+- [ ] (If necessary) Secure a release any and all family buildpacks that contain the new implementation buildpack
```

Comment:
Could we add clarity around when this is necessary?

I might expect that this is always necessary and that automation will usually create the bump PR into the language family and that the maintainers will take over from there. 

Or is the implication that if you consume the feature via the language family you should take ownership over getting the language family bumped?

Reply:
I think that it could be a feature change in the language family potentially (i.e. you add a new oder grouping or expand the utilities buildpacks) by cutting securing the release of the buildpack you are also secure a release of the family buildpack (because they are one in the same).

This might be a bit pedantic and unclear though so let me know if you would prefer another wording.